chameleon/views.py

```diff
@@ -231,7 +232,8 @@ def dashboard(request):
             proj.__dict__['nickname'] = extras.nickname
         except ProjectExtras.DoesNotExist:
             project_nickname = None
-
+        
+        proj = project_allocations_format_mapping(proj)
```

Comment:
How could we make the decision of "fetch allocations from portal... or not" be based on the incoming request? I'd like to make it so we can roll this out in "dark launch", meaning most users don't interact with it at all.

I suggest you make this mapping thing a class, which is instantialized with a request, e.g.

```python
class ProjectAllocationMapper:
    def __init__(self, request):
        self.allocations_from_db = self._wants_db_allocations(request)
    
    def map(self, project):
        if self.allocations_from_db:
            # Do your fetching and mapping of allocations
            some_logic
        else:
            return project
```

Then in your controller (`dashboard`), you can do like:

```python
mapper = ProjectAllocationsMapper(request)

proj = mapper.map(proj)
```

Then we can vary the logic of `_wants_db_allocations` to look at a cookie, or perhaps the logged in user ID, or something else entirely (like hash the user ID and then do some bucketing such that we roll out to 50% of users, then 100%)